### PR TITLE
Improve seat allocation on semicircle plot

### LINKED ISSUE #20: Accurate semicircle parliament plot allocations
The semicircle version of the [parliament plot](https://github.com/andrewtavis/poli-sci-kit/blob/main/src/poli_sci_kit/plot/parliament.py) at times doesn't put all the seats for a group in the same area. There is potentially a better method of allocating the scatterplot points in that function, or break points could be put in given parliament sizes to assure that groups have all their seats in one area of the plot.

Fixes for this would specifically need to target [poli_sci_kit.utils.gen_parl_points](https://github.com/andrewtavis/poli-sci-kit/blob/main/src/poli_sci_kit/utils.py).

### PR DESCRIPTION
**Pull request by gnegrelli**
Fix #20 and #30.

Improve seat allocation by grouping points in the scatterplot based on the angle between them and the origin (0, 0).

Also, raise an exception when seats cannot be properly allocated into the number of rows desired.

Allow users to tweak plot via kwargs.

### CODE REVIEW

**Comment by andrewtavis** [comment]
After this we should definitely do a release. Let's say `v1.1.0` as this is a non-trivial improvement. I'm wondering if we need to update anything else though? What's your Python version? I think that there aren't too many dependencies for this that would cause problems with us making the jump to 3.9/10/11. Maybe another issue?

**Comment by andrewtavis** [comment]
A note to myself as well: one of the shields on readthedocs is broken via the breaking changes that went through in shields.io a while ago :)

**Comment by gnegrelli** [comment]
> After this we should definitely do a release. Let's say `v1.1.0` as this is a non-trivial improvement. I'm wondering if we need to update anything else though? What's your Python version? I think that there aren't too many dependencies for this that would cause problems with us making the jump to 3.9/10/11. Maybe another issue?

I agree with the release. I don't see what else needs updating, but I've only worked on the plotting part of the package. It would be nice to test the remaining code/examples to see if it's all working well. 

I've used Python 3.10. The dependencies/requirements didn't gave me trouble at all, so I guess it would be safe to bump the version.

**Comment by andrewtavis** [comment]
Ok, I'll play around a bit post merge and then we'll get it onto PyPI 😊

**Comment by andrewtavis** [comment]
Hey @gnegrelli 👋 Just FYI I'm in the midst of a very busy week. I'll try to get to this Sunday or early next week when I'm on vacation :)

**Comment by gnegrelli** [comment]
No worries, @andrewtavis. Take your time.

**Comment by andrewtavis** [comment]
Let's not worry about the new codecov issues, @gnegrelli :) Getting to this this evening, hopefully.

File: src/poli_sci_kit/utils.py [hunk 1]
```diff
@@ -141,17 +141,20 @@ def arc_coordinates(r, seats):
...
 
             # Broadcast angles to their corresponding coordinates.
             x_coordinates = list(r * np.cos(angles))
             y_coordinates = list(r * np.sin(angles))
 
-            return x_coordinates, y_coordinates
+            return x_coordinates, y_coordinates, list(angles)
 
-        xs, ys = [], []
+        # Store point coordinates (x, y) and their angles with origin (0, 0)
```
Comment (andrewtavis) [inline src/poli_sci_kit/utils.py:151]: Minor nitpick, with Python it's customary that full line comments are complete sentences with punctuation :) For in-line comments there's no punctuation and the first word should not be capitalized 😊 Just FYI, I can get these!

File: src/poli_sci_kit/utils.py [hunk 2]
```diff
@@ -171,66 +174,39 @@ def arc_coordinates(r, seats):
         ]  # greater shift for higher rows for more equal spacing
         seats_per_row = [rs + seats_shift[i] for i, rs in enumerate(row_seats)]
 
+        if any(seats <= 0 for seats in seats_per_row):
+            raise Exception(f'Cannot allocate {total_seats} seats into {num_rows} rows. Try a smaller number of rows.')
```
Comment (andrewtavis) [inline src/poli_sci_kit/utils.py:178]: Note to self that this should be double quotes :)

File: src/poli_sci_kit/utils.py [hunk 3]
```diff
@@ -171,66 +174,39 @@ def arc_coordinates(r, seats):
...
             item for sublist in row_position_indexes for item in sublist
         ]
 
-        # Index the group and deplete a copy of allocations at its location.
-        group_index = 0
-        seats_to_allocate = allocations.copy()
-        row_index = 0
+        # Generate list of seat labels
+        seat_labels = []
+        for n_seats, label in zip(allocations, labels):
```
Comment (andrewtavis) [inline src/poli_sci_kit/utils.py:202]: Nice :)

**Review by andrewtavis — APPROVED** [review]
All looks really really good, @gnegrelli! I'll bring this in and then we can write a test for #30 and expand the example out with something that shows off the new abilities of the parliament plot a bit more! Let's shoot to have the new release done by EOW. I can work on the example, and if you want to do the test then that'd be great!

**Pull request merged**